packages/core/src/utils/onErrorResumeNext.ts

```diff
@@ -0,0 +1,15 @@
+export default function onErrorResumeNext<T>(fn: () => T): T | undefined;
+export default function onErrorResumeNext<T, U = T>(fn: () => T, defaultValue: U): T | U;
+
+/**
+ * Calls the function and returns its return value.
+ *
+ * If the function failed, return default value if specified, otherwise, `undefined`.
+ */
+export default function onErrorResumeNext<T>(fn: () => T, defaultValue?: T): T | undefined {
```

Comment:
I wonder if you can do something like this so the type only indicates `undefined` if the default value is not provided....
```ts
export default function onErrorResumeNext<T>(
  fn: () => T,
  defaultValue?: T
): typeof defaultValue extends undefined ? T | undefined : T;
```